Formula/weechat.rb

```diff
@@ -26,7 +35,7 @@ class Weechat < Formula
   depends_on "lua" => :optional
   depends_on "perl" => :optional
   depends_on "python" => :optional
-  depends_on "ruby" => :optional if MacOS.version <= :sierra
```

Comment:
why?

Reply:
You didn't really explain in the other PR _why_ you'd capped this below Sierra, so, you tell me? 🙈

Reply:
The problem remains either way. Homebrew's Ruby is at 2.5.0, which `weechat` stable has no idea exists before this PR. But I can find no fault building against Ruby on High Sierra other than that.

Reply:
The minimum version was set to 2.1

Reply:
I'll look around but this proved an exceptionally stubborn issue in the past.

Reply:
```
RUBY_EXECUTABLE:FILEPATH=/usr/bin/ruby
RUBY_INCLUDE_DIRS:PATH=/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0
RUBY_LIB:FILEPATH=/usr/lib/libruby.dylib
```

Reply:
looks like `-DRUBY_EXECUTABLE=/usr/bin/ruby` is sufficient

Reply:
Alas, it isn't.

```
==> Summary
🍺  /usr/local/Cellar/weechat/2.0.1: 40 files, 5.9MB, built in 24 seconds
~> otool -L /usr/local/Cellar/weechat/2.0.1/lib/weechat/plugins/ruby.so
/usr/local/Cellar/weechat/2.0.1/lib/weechat/plugins/ruby.so:
	/usr/local/opt/ruby/lib/libruby.2.5.dylib (compatibility version 2.5.0, current version 2.5.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
```

Reply:
```diff
diff --git a/Formula/weechat.rb b/Formula/weechat.rb
index fabaa5dd1a..40ff809fdd 100644
--- a/Formula/weechat.rb
+++ b/Formula/weechat.rb
@@ -35,7 +35,7 @@ class Weechat < Formula
   depends_on "lua" => :optional
   depends_on "perl" => :optional
   depends_on "python" => :optional
-  depends_on "ruby" => :optional
+  depends_on "ruby" => :optional if MacOS.version <= :sierra
   depends_on "curl" => :optional
 
   def install
@@ -49,9 +49,14 @@ class Weechat < Formula
       args << "-DCMAKE_BUILD_TYPE=Debug"
     end
 
+    if build.without? "ruby"
+      args << "-DENABLE_RUBY=OFF"
+    elsif build.with?("ruby") && MacOS.version >= :sierra
+      args << "-DRUBY_EXECUTABLE=/usr/bin/ruby"
+    end
+
     args << "-DENABLE_LUA=OFF" if build.without? "lua"
     args << "-DENABLE_PERL=OFF" if build.without? "perl"
-    args << "-DENABLE_RUBY=OFF" if build.without? "ruby"
     args << "-DENABLE_ASPELL=OFF" if build.without? "aspell"
     args << "-DENABLE_TCL=OFF" if build.without? "tcl"
     args << "-DENABLE_PYTHON=OFF" if build.without? "python"
```

Reply:
Throwing `-DRUBY_LIB=/usr/lib/libruby.dylib` at it too fixes that.